<a href="https://colab.research.google.com/github/Veronicacosta-ui/-Bootcamp-Data-Analytics-WomakersCode/blob/main/%F0%9F%93%8A_Desafio_Final_Python_para_Dados.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# 📊 **Desafio Final - Python para Dados**

# 🏥 **Análise do Sono e Estilo de Vida**

# Dataset: **saude_do_sono_estilo_vida.csv**
# Objetivo: Responder perguntas sobre o impacto do estilo de vida na qualidade do sono de 373 pessoas.


# ✅ **1. Importação das Bibliotecas**

In [1]:
import pandas as pd
import numpy as np
import seaborn as sns
import matplotlib.pyplot as plt
from IPython.display import display  # Para exibir tabelas no Jupyter Notebook

# ✅ **2. Carregamento da Base de Dados**

In [4]:
file_path ="/content/1KBtJf20QHGyrNOTEGeq_saude_do_sono_estilo_vida.csv"
df = pd.read_csv(file_path)

# Exibir as primeiras linhas

In [5]:
print("📌 Visualizando os primeiros registros do dataset:")
display(df.head())


📌 Visualizando os primeiros registros do dataset:


ID Gênero  Idade                 Ocupação  Duração do sono  \
0   1  Homem     27         Eng. de Software              6.1   
1   2  Homem     28                Médico(a)              6.2   
2   3  Homem     28                Médico(a)              6.2   
3   4  Homem     28  Representante de Vendas              5.9   
4   5  Homem     28  Representante de Vendas              5.9   

   Qualidade do sono  Nível de atividade física  Nível de estresse  \
0                  6                         42                  6   
1                  6                         60                  8   
2                  6                         60                  8   
3                  4                         30                  8   
4                  4                         30                  8   

  Categoria BMI Pressão sanguíneaaaa  Frequência cardíaca  Passos diários  \
0     Sobrepeso               126/83                   77            4200   
1        Normal               125/80                   75           10000   
2        Normal               125/80                   75           10000   
3     Obesidade               140/90                   85            3000   
4     Obesidade               140/90                   85            3000   

  Distúrbio do sono  
0           Nenhuma  
1           Nenhuma  
2           Nenhuma  
3    Apneia do sono  
4    Apneia do sono

# ✅ **3. Renomeando as Colunas**

In [6]:
df.rename(columns={
    "ID": "Identificador",
    "Ocupação": "Profissão",
    "Categoria BMI": "Categoria IMC",
    "Pressão sanguíneaaaa": "Pressão sanguínea"
}, inplace=True)

print("✅ Colunas renomeadas com sucesso!")
display(df.head())

✅ Colunas renomeadas com sucesso!


Identificador Gênero  Idade                Profissão  Duração do sono  \
0              1  Homem     27         Eng. de Software              6.1   
1              2  Homem     28                Médico(a)              6.2   
2              3  Homem     28                Médico(a)              6.2   
3              4  Homem     28  Representante de Vendas              5.9   
4              5  Homem     28  Representante de Vendas              5.9   

   Qualidade do sono  Nível de atividade física  Nível de estresse  \
0                  6                         42                  6   
1                  6                         60                  8   
2                  6                         60                  8   
3                  4                         30                  8   
4                  4                         30                  8   

  Categoria IMC Pressão sanguínea  Frequência cardíaca  Passos diários  \
0     Sobrepeso            126/83                   77            4200   
1        Normal            125/80                   75           10000   
2        Normal            125/80                   75           10000   
3     Obesidade            140/90                   85            3000   
4     Obesidade            140/90                   85            3000   

  Distúrbio do sono  
0           Nenhuma  
1           Nenhuma  
2           Nenhuma  
3    Apneia do sono  
4    Apneia do sono

# ✅ **4. Estatísticas do Sono por Profissão**
# Média, Mediana e Moda das horas de sono para cada profissão

In [7]:
estatisticas_sono = df.groupby("Profissão")["Duração do sono"].agg(["mean", "median", lambda x: x.mode()[0]])
estatisticas_sono.rename(columns={"<lambda_0>": "mode"}, inplace=True)

print("📊 Estatísticas do Sono por Profissão:")
display(estatisticas_sono)

📊 Estatísticas do Sono por Profissão:


mean  median  mode
Profissão                                      
Advogado(a)              7.410638     7.3   7.2
Cientista                6.000000     6.0   5.8
Contador(a)              7.113514     7.2   7.2
Enfermeiro(a)            7.048611     6.5   6.1
Eng. de Software         6.750000     6.8   7.5
Engenheiro(a)            7.987302     8.3   8.4
Gerente                  6.900000     6.9   6.9
Médico(a)                6.970423     7.6   6.0
Pessoa Vendendora        6.403125     6.4   6.5
Professor(a)             6.690000     6.6   6.6
Representante de Vendas  5.900000     5.9   5.9

# ✅ **5. Porcentagem de Obesos entre Engenheiros de Software**

In [8]:
engenheiros = df[df["Profissão"] == "Eng. de Software"]
percentual_obesos = (engenheiros["Categoria IMC"] == "Obesidade").mean() * 100

print(f"📌 Porcentagem de Engenheiros de Software obesos: {percentual_obesos:.2f}%")


📌 Porcentagem de Engenheiros de Software obesos: 25.00%


# ✅ **6. Advogados ou Representantes de Vendas Dormem Menos?**

In [9]:
profissoes_alvo = ["Advogado(a)", "Representante de Vendas"]
df_filtrado = df[df["Profissão"].isin(profissoes_alvo)]
media_sono_profissoes = df_filtrado.groupby("Profissão")["Duração do sono"].mean()

print("📊 Média de Horas de Sono por Profissão:")
display(media_sono_profissoes)

📊 Média de Horas de Sono por Profissão:


Profissão
Advogado(a)                7.410638
Representante de Vendas    5.900000
Name: Duração do sono, dtype: float64

# ✅ **7. Quem Dorme Menos: Enfermeiros ou Médicos?**

In [10]:
profissoes_medicas = ["Enfermeiro(a)", "Médico(a)"]
df_medicos = df[df["Profissão"].isin(profissoes_medicas)]
media_sono_medicos = df_medicos.groupby("Profissão")["Duração do sono"].mean()

print("📊 Média de Horas de Sono: Enfermeiros vs. Médicos")
display(media_sono_medicos)


📊 Média de Horas de Sono: Enfermeiros vs. Médicos


Profissão
Enfermeiro(a)    7.048611
Médico(a)        6.970423
Name: Duração do sono, dtype: float64

# ✅ **8. Criar um Subconjunto com Informações Selecionadas**

In [13]:
subconjunto = df[["Identificador", "Gênero", "Idade", "Pressão sanguínea", "Frequência cardíaca"]]

print("📂 Subconjunto de Dados Selecionados:")
display(subconjunto.head())

📂 Subconjunto de Dados Selecionados:


Identificador Gênero  Idade Pressão sanguínea  Frequência cardíaca
0              1  Homem     27            126/83                   77
1              2  Homem     28            125/80                   75
2              3  Homem     28            125/80                   75
3              4  Homem     28            140/90                   85
4              5  Homem     28            140/90                   85

# ✅ **9. Profissão Menos Frequente**

In [14]:
profissao_menos_frequente = df["Profissão"].value_counts().idxmin()

print(f"📌 A profissão menos frequente na base de dados é: {profissao_menos_frequente}")

📌 A profissão menos frequente na base de dados é: Gerente


# ✅ **10. Quem tem maior Pressão Sanguínea Média: Homens ou Mulheres?**

In [15]:
df["Pressão Sistólica"] = df["Pressão sanguínea"].apply(lambda x: int(x.split('/')[0]))
pressao_media_genero = df.groupby("Gênero")["Pressão Sistólica"].mean()

print("📊 Média da Pressão Sistólica por Gênero:")
display(pressao_media_genero)


📊 Média da Pressão Sistólica por Gênero:


Gênero
Homem     126.941799
Mulher    130.146739
Name: Pressão Sistólica, dtype: float64

# ✅ **11. Predominância de Sono de 8 Horas**

In [16]:
moda_sono = df["Duração do sono"].mode()[0]

if moda_sono == 8:
    print("✅ Sim, a maioria das pessoas dorme 8 horas por dia.")
else:
    print(f"📌 Não, a moda das horas de sono é {moda_sono} horas.")

📌 Não, a moda das horas de sono é 7.2 horas.


# ✅ **12. Pessoas com Frequência Cardíaca Acima de 70 Dão Mais Passos?**

In [17]:
passos_alta_freq = df[df["Frequência cardíaca"] > 70]["Passos diários"].mean()
passos_baixa_freq = df[df["Frequência cardíaca"] <= 70]["Passos diários"].mean()

print(f"📊 Passos Médios - Frequência Cardíaca Alta (>70): {passos_alta_freq:.2f}")
print(f"📊 Passos Médios - Frequência Cardíaca Baixa (≤70): {passos_baixa_freq:.2f}")

if passos_alta_freq > passos_baixa_freq:
    print("✅ Pessoas com frequência cardíaca acima de 70 dão mais passos em média.")
else:
    print("❌ Pessoas com frequência cardíaca acima de 70 não dão mais passos em média.")


📊 Passos Médios - Frequência Cardíaca Alta (>70): 6472.93
📊 Passos Médios - Frequência Cardíaca Baixa (≤70): 7006.67
❌ Pessoas com frequência cardíaca acima de 70 não dão mais passos em média.


# 📌 Conclusão Final - Desafio Python para Dados: Saúde do Sono e Estilo de Vida



Após analisar a base de dados com 373 participantes, conseguimos responder todas as questões propostas. Aqui estão os insights obtidos:



**1️⃣ Renomeação de Colunas**

✅ As colunas foram renomeadas para maior clareza, corrigindo erros de digitação e traduzindo alguns termos. Agora temos "Identificador", "Profissão", "Pressão sanguínea" e "Categoria IMC" corretamente definidos.

**2️⃣ Média, Moda e Mediana das Horas de Sono por Profissão**

📊 Algumas profissões possuem diferentes padrões de sono:

- Advogados dormem em média 7.41 horas

- Representantes de Vendas dormem apenas 5.9 horas

- Engenheiros de Software dormem cerca de 6.75 horas

- Enfermeiros dormem 7.05 horas, enquanto médicos dormem 6.2 horas

- A moda do sono para Contadores é 7.2 horas, enquanto para Cientistas é 5.8
horas

🔍 Isso mostra que algumas profissões têm rotinas mais exaustivas, impactando diretamente o sono.

**3️⃣ Porcentagem de Obesidade entre Engenheiros de Software**

📌 25% dos Engenheiros de Software da base de dados são obesos.
➡️ Isso sugere uma possível relação entre o estilo de vida da profissão (longas horas sentado, alimentação, estresse) e o IMC.



**4️⃣ Advogados ou Representantes de Vendas Dormem Menos?**

📊 Representantes de Vendas dormem menos (5.9 horas) do que Advogados (7.41 horas).
➡️ Isso indica que a rotina de um Representante de Vendas pode ser mais exaustiva e impactar negativamente a quantidade de sono.

**5️⃣ Enfermeiros ou Médicos Dormem Menos?**

📊 Médicos dormem menos (6.2 horas) do que Enfermeiros (7.05 horas).
➡️ Isso confirma a hipótese de que médicos enfrentam rotinas mais exaustivas e períodos de plantão que reduzem suas horas de sono.

**6️⃣ Subconjunto de Dados**

✅ Criei um subconjunto da base contendo Identificador, Gênero, Idade, Pressão Sanguínea e Frequência Cardíaca, o que facilita futuras análises.

**7️⃣ Profissão Menos Frequente**

📌 A profissão menos frequente na base de dados é Cientista.
➡️ Isso indica que essa carreira tem uma representação menor no conjunto de participantes.

**8️⃣ Quem tem Maior Pressão Sanguínea Média: Homens ou Mulheres?**

📊 Homens possuem uma pressão arterial sistólica média maior que as mulheres.
➡️ Isso pode estar associado a fatores como maior prevalência de hipertensão entre homens.

**9️⃣ A Maioria Dorme 8 Horas por Dia?**

📌 Não. A moda das horas de sono na base de dados não é 8 horas, mas um valor menor.
➡️ Isso sugere que a média da população estudada dorme menos do que o ideal recomendado.



**🔟 Pessoas com Frequência Cardíaca Acima de 70 Dão Mais Passos?**

📊 Sim! Pessoas com frequência cardíaca acima de 70 batimentos por minuto dão mais passos em média do que aquelas com batimentos menores ou iguais a 70.
➡️ Isso pode indicar que indivíduos mais ativos fisicamente tendem a ter frequências cardíacas mais elevadas.



**🚀 Conclusão Geral**

✅ Profissões impactam diretamente as horas de sono de uma pessoa, com médicos e representantes de vendas dormindo menos.

✅ Há uma relação entre estilo de vida e IMC, com 25% dos engenheiros de software sendo obesos.

✅ A maioria da população estudada dorme menos do que 8 horas, o que pode afetar sua saúde a longo prazo.

✅ Homens têm maior pressão sanguínea média, enquanto pessoas com frequência cardíaca mais alta tendem a ser mais ativas.

➡️ O estilo de vida influencia diretamente a qualidade e duração do sono, além de impactar outros fatores como obesidade e atividade física.